src/components/pages/Users/GroupsTab/GroupsTab.test.tsx

```diff
@@ -0,0 +1,205 @@
+import { NotificationSeverity } from "@canonical/react-components";
+import { screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { http, delay } from "msw";
+import { setupServer } from "msw/node";
+
+import {
+  getGetGroupsMockHandler,
+  getGetGroupsResponseMock,
+} from "api/groups/groups.msw";
+import {
+  getPatchIdentitiesItemGroupsMockHandler,
+  getGetIdentitiesItemGroupsMockHandler,
+  getGetIdentitiesItemGroupsResponseMock,
+  getPatchIdentitiesItemGroupsMockHandler400,
+  getGetIdentitiesItemGroupsMockHandler400,
+} from "api/identities/identities.msw";
+import { Label as GroupsPanelFormLabel } from "components/GroupsPanelForm";
+import { hasNotification, hasSpinner, renderComponent } from "test/utils";
+import { Endpoint } from "types/api";
+import urls from "urls";
+
+import GroupsTab from "./GroupsTab";
+import { Label } from "./types";
+
+const path = urls.users.user.groups(null);
+const url = urls.users.user.groups({ id: "user1" });
+
+const mockApiServer = setupServer(
+  getPatchIdentitiesItemGroupsMockHandler(),
+  getGetIdentitiesItemGroupsMockHandler(
+    getGetIdentitiesItemGroupsResponseMock({
+      data: [
+        { id: "group123", name: "group1" },
+        { id: "group234", name: "group2" },
+      ],
+    }),
+  ),
+  getGetGroupsMockHandler(
+    getGetGroupsResponseMock({
+      data: [
+        { id: "group123", name: "group1" },
```

Comment:
I think we could potentially use the newly added mock function for groups. https://github.com/canonical/rebac-admin/blob/main/src/mocks/groups.ts

Reply:
Thanks! Done.